src/BaseExtension.php

```diff
@@ -450,7 +450,7 @@ public function addSnippet($location, $callback, $extraparameters = [])
             $callback = (string) $callback;
         }
 
-        $this->app['asset.queue.snippet']->add($location, $callback, $this->getName(), (array) $extraparameters);
+        $this->app['asset.queue.snippet']->add($location, [$this, $callback], $this->getName(), (array) $extraparameters);
     }
```

Comment:
I had that at first, but since `addTwigFunction` and `addTwigFilter` don't check for this either, I thought I wouldn't have to here either. I'll fix this, and add checks at the other two functions as well.